File: src/my_project/medium_problems/from101to150/group_anagrams.py
from typing import List 
import collections

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
    
        m = collections.defaultdict(list)
        for s in strs:
            m["".join(sorted(s))].append(s)
        return [v for v in m.values()]
    
    
a = 'cba'
b = sorted(a)
print(b)

File: src/my_project_typescript/medium_problems_typescript/from1to50/groupAnagrams.ts
function groupAnagrams(strs: string[]): string[][] {



    let dicitonarySolution: {[key:string]: Array<string>} = {}

    for (let word of strs){
        dicitonarySolution[word.split('').sort().join('')] = []
    }

    for (let word of strs){
        dicitonarySolution[word.split('').sort().join('')].push(word)
    }

    let solution: Array<Array<string>> = []

    for (let value in dicitonarySolution){
        solution.push(dicitonarySolution[value])
    }

    return solution

};


let sample: Array<string> = ['c','b','a']
let sampleTwo: Array<string> = sample.sort()
let a: string = 'dcba'
console.log(a.split(''))

console.log(sampleTwo)

console.log(groupAnagrams(["eat","tea","tan","ate","nat","bat"]))